### PR TITLE
REST

### PR DESCRIPTION
**Pull request by Danvace**

### CODE REVIEW

File: src/main/java/ua/lviv/iot/algo/part1/lab1/repository/HornetRepository.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package ua.lviv.iot.algo.part1.lab1.repository;
+
+import org.springframework.stereotype.Repository;
+import ua.lviv.iot.algo.part1.lab1.models.Hornet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
```
Comment (zveres-ss) [inline src/main/java/ua/lviv/iot/algo/part1/lab1/repository/HornetRepository.java]: Please check which classes should be marked with @Repository
the map should be used in class with @Service annotation

File: src/main/java/ua/lviv/iot/algo/part1/lab1/models/Hornet.java [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
         setNumberOfLegs(numberOfLegs);
         setHasWings(hasWings);
         setIsDangerous(isDangerous);
         setName(name);
         setIsSleeping(isSleeping);
+
+    }
+
+    public Hornet() {
+        this.id = Insect.idCreate();
```
Comment (zveres-ss) [inline src/main/java/ua/lviv/iot/algo/part1/lab1/models/Hornet.java]: redundant changes. You should NEVER assign id by yourself during object creation

File: src/main/java/ua/lviv/iot/algo/part1/lab1/controller/HornetController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @RequestMapping(path = "/post")
+    @PostMapping()
+    public void postHornet() {
+        var hornet = new Hornet();
+        hornetService.addHornet(hornet);
+    }
+
+    @RequestMapping(path = "/edit")
```
Comment (zveres-ss) [inline src/main/java/ua/lviv/iot/algo/part1/lab1/controller/HornetController.java]: please use REST approach for path creation as it is explained here: https://youtu.be/lq2NSBTxp_8?t=1676

File: src/main/java/ua/lviv/iot/algo/part1/lab1/models/Hornet.java [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
+import lombok.Getter;
+import lombok.ToString;
+import org.junit.Test;
 import ua.lviv.iot.algo.part1.lab1.Hibernationality;
-
+@ToString(callSuper = true)
+@Getter
 public class Hornet extends Insect implements Hibernationality {
 
     public static final String HEADERS = "isOldQueen, isQueen";
```
Comment (zveres-ss) [inline src/main/java/ua/lviv/iot/algo/part1/lab1/models/Hornet.java]: you added new field - your headers and CSV values should reflect this change

**Review by zveres-ss — CHANGES_REQUESTED** [review]

File: src/main/java/ua/lviv/iot/algo/part1/lab1/controller/HornetController.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import ua.lviv.iot.algo.part1.lab1.models.Hornet;
+import ua.lviv.iot.algo.part1.lab1.service.HornetService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/Hornets")
```
Comment (zveres-ss) [inline src/main/java/ua/lviv/iot/algo/part1/lab1/controller/HornetController.java]: `@RequestMapping(path = "/hornets")`

File: src/main/java/ua/lviv/iot/algo/part1/lab1/service/HornetService.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    public void addHornet(final Hornet hornet) {
+        hornetMap.put(this.nextAvailable++, hornet);
+    }
+
+    public void deleteHornet(final Integer id) {
+        hornetMap.remove(id);
+    }
+
+    public Hornet getHornet(final Integer id) {
+        return new Hornet(hornetMap.get(id),id);
```
Comment (zveres-ss) [inline src/main/java/ua/lviv/iot/algo/part1/lab1/service/HornetService.java]: why you're creating new object instead of reusing existing?

**Review by zveres-ss — CHANGES_REQUESTED** [review]

File: src/main/java/ua/lviv/iot/algo/part1/lab1/service/HornetService.java (hunk 6 above)
Comment (Danvace) [reply src/main/java/ua/lviv/iot/algo/part1/lab1/service/HornetService.java]: so that vulnerabilities do not arise

**Pull request merged**